WcaOnRails/app/views/competitions/index.html.erb

```diff
@@ -1,17 +1,99 @@
 <% provide(:title, 'Competitions') %>
 
 <div class="container">
-  <h1><%= yield(:title) %></h1>
-
-  <%= link_to "New competition", new_competition_path %>
-  <table class="table">
-    <% @competitions.each do |competition| %>
-      <tr>
-        <td><%= competition.id %></td>
-        <td><%= competition.name %></td>
-        <td><%= link_to "Edit", admin_edit_competition_path(competition) %></td>
-        <td><%= link_to "Results", "/results/competition.php?competitionId=#{competition.id}" %></td>
-      </tr>
+  <h2><%= yield(:title) %></h2>
+
+  <div id="scroller-anchor"></div>
+  <div id="scroller">
+    <%= form_tag competitions_path, method: :get, class: "competition-select form-inline" do %>
+      <%= render 'index_form' %>
+    <% end %>
+  </div>
+    <% if @competitions.length == 0 %>
+      <div class="alert alert-warning">No competitions found!</div>
+    <% else %>
+      <% if params[:display] == "List" %>
+        <div class="col-12" style="z-index: -1;">
```

Comment:
The z-index is to prevent the tables from scrolling over the form.
I'll make it a class.

Didn't understand the part about "need to be in a row"...
I noticed the alignment was off, but it looked the same as your example, so I guessed it was correct.